### PR TITLE
feat: typescript sdk filter documentation update

### LINKED ISSUE #381: TypeScript: Update `filters` documentation
There was a breaking change to filters introduced in https://github.com/tigrisdata/tigris-client-ts/pull/322 and here are the few places where filters are being used:

### TypeScript database

- https://github.com/tigrisdata/tigris-docs/blob/main/docs/sdkstools/typescript/database/delete.mdx
- https://github.com/tigrisdata/tigris-docs/blob/main/docs/sdkstools/typescript/database/insert.mdx
- https://github.com/tigrisdata/tigris-docs/blob/main/docs/sdkstools/typescript/database/query.mdx
- https://github.com/tigrisdata/tigris-docs/blob/main/docs/sdkstools/typescript/database/search.mdx
- https://github.com/tigrisdata/tigris-docs/blob/main/docs/sdkstools/typescript/database/update.mdx

### TypeScript search

- https://github.com/tigrisdata/tigris-docs/blob/main/docs/sdkstools/typescript/search/fulltext-search.mdx
- https://github.com/tigrisdata/tigris-docs/blob/main/docs/sdkstools/typescript/search/vector-search.mdx

^This is not a comprehensive list, we might have to do a grep across the codebase

### PR DESCRIPTION
**Pull request by akhill10**
- Closes #381 
https://github.com/tigrisdata/tigris-client-ts/pull/322

### CODE REVIEW

File: docs/sdkstools/typescript/_events/query.mdx [hunk 1]
```diff
@@ -17,13 +17,13 @@ import readAllSource from "!!raw-loader!../_snippets/query_topic/_readAll";
...
 
 export const filterSyntax = {
-  eq: "SelectorFilterOperator.EQ",
-  lt: "SelectorFilterOperator.LT",
-  lte: "SelectorFilterOperator.LTE",
-  gt: "SelectorFilterOperator.GT",
-  gte: "SelectorFilterOperator.GTE",
-  or: "LogicalOperator.OR",
-  and: "LogicalOperator.AND",
+  eq: "value",
```
Comment (akhill10) [inline docs/sdkstools/typescript/_events/query.mdx:20]: @adilansari  Need help here. 
With the recent filter update, we aren't considering the EQ operator. We are passing direct value to the key. For example:
 Before version:
```
 const cancelledOrdersFilter = {
  op: SelectorFilterOperator.EQ,
  fields: {
    orderStatus: "cancelled"
  }
};

```
Latest version: 
```
 const cancelledOrdersFilter = {
    orderStatus: "cancelled",
};
```

So what's the best term over here?

File: docs/sdkstools/typescript/search/fulltext-search.mdx [hunk 2]
```diff
@@ -231,14 +228,13 @@ const query: SearchQuery<Catalog> = {
 
 The following filter operators are supported:
 
-- `SelectorFilterOperator.EQ` - Matches documents where the field value is equal to the provided value.
```
Comment (akhill10) [inline docs/sdkstools/typescript/search/fulltext-search.mdx:234]: @adilansari Need help here
I removed this point, we are not supporting this operator, Is it okay? Or should we add something here ?

File: docs/sdkstools/typescript/_events/query.mdx [hunk 3]
```diff
@@ -17,13 +17,13 @@ import readAllSource from "!!raw-loader!../_snippets/query_topic/_readAll";
...
+  lt: "$lt",
+  lte: "$lte",
+  gt: "$gt",
+  gte: "$gte",
+  or: "$or",
+  and: "$and",
```
Comment (leggetter) [inline docs/sdkstools/typescript/_events/query.mdx:26]: Will we also support `$note`, `$contains` etc. See https://github.com/tigrisdata/tigris/blob/42b9f8a0529f398cc7267ad41f5f5f3f0c659b1c/query/filter/comparison.go#L28-L35

Comment (adilansari) [reply docs/sdkstools/typescript/_events/query.mdx:26]: Yeah, they are not in TypeScript client yet. Added a separate issue to implement them:
https://github.com/tigrisdata/tigris-client-ts/issues/325

File: docs/sdkstools/typescript/search/fulltext-search.mdx (hunk 2 above)
Comment (adilansari) [reply docs/sdkstools/typescript/search/fulltext-search.mdx:234]: We have a [`$eq` operator](https://github.com/tigrisdata/tigris-client-ts/blob/93d099072219f60879b87e3779c8eb6d82ec92a9/src/types.ts#L765), just that its optional, right?

If not, can we make these equivalent?

```ts
const nameFilter: Filter<IUser1> = { name: "Jim" };
```

should be equivalent to:

```ts
const nameFilter: Filter<IUser1> = {
			name: {
				$eq: "Jim",
			},
		};
```

And then,
```
- `$eq` - Matches documents where the field value is equal to the provided value.
```

**Comment by adilansari** [comment]
@akhill10 thank you for resolving this big feature.

File: docs/sdkstools/typescript/_events/query.mdx (hunk 1 above)
Comment (adilansari) [reply docs/sdkstools/typescript/_events/query.mdx:20]: We can accept `$eq` operator as an optional, check my other comment.

File: docs/sdkstools/typescript/search/fulltext-search.mdx (hunk 2 above)
Comment (akhill10) [reply docs/sdkstools/typescript/search/fulltext-search.mdx:234]: Yeah, it's just optional. Server accepts both of these variations, right?

Comment (akhill10) [reply docs/sdkstools/typescript/search/fulltext-search.mdx:234]: Changes are done.

**Review by adilansari — APPROVED** [review]

**Pull request merged**